pkg/controllermanager/controller/event/reconciler.go

```diff
@@ -19,47 +19,35 @@ import (
 	"fmt"
 	"time"
 
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/tools/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/ratelimiter"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
-	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
 )
 
 var nowFunc = time.Now
 
-func (c *Controller) enqueueEvent(obj interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(obj)
-	if err != nil {
-		c.log.Error(err, "Couldn't get key for object", "object", obj)
-		return
-	}
-	c.eventQueue.Add(key)
-}
-
-// NewEventReconciler creates a new instance of a reconciler which reconciles Events.
-func NewEventReconciler(gardenClient client.Client, cfg *config.EventControllerConfiguration) reconcile.Reconciler {
-	return &eventReconciler{
-		gardenClient: gardenClient,
-		cfg:          cfg,
-	}
-}
+// Reconciler reconciles Event.
+type Reconciler struct {
+	Client client.Client
+	Config config.EventControllerConfiguration
 
-type eventReconciler struct {
-	gardenClient client.Client
-	cfg          *config.EventControllerConfiguration
+	// RateLimiter allows limiting exponential back off for testing purposes
+	RateLimiter ratelimiter.RateLimiter
```

Comment:
There is no exponential backoff in this controller during normal operation, so you could remove it.